src/tree.rs

```diff
@@ -194,25 +194,6 @@ impl MerkleHashOrchard {
     }
 }
 
-/// This instance should only be used for hash table key comparisons.
-impl std::cmp::PartialEq for MerkleHashOrchard {
-    fn eq(&self, other: &Self) -> bool {
-        self.0.ct_eq(&other.0).into()
-    }
-}
-
-/// This instance should only be used for hash table key comparisons.
-impl std::cmp::Eq for MerkleHashOrchard {}
-
-/// This instance should only be used for hash table key hashing.
-impl std::hash::Hash for MerkleHashOrchard {
-    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
-        <Option<pallas::Base>>::from(self.0)
-            .map(|b| b.to_bytes())
-            .hash(state)
-    }
-}
-
```

Comment:
Why were these removed?

Reply:
They're duplicative of the ones that are now derived; pallas::Base now has `Eq` and `Ord` implementations that delegate to the constant-time versions.

Reply:
- I confimed this is the case for `PartialEq, Eq`.
- `PartialOrd, Ord` is not implemented in constant time in `pasta_curves`, but it does implement lexicographical ordering of the encodings, which seems reasonable for `MerkleHashOrchard` to inherit (in that we aren't trying to sort them as integers).
- `Hash` is still being removed, preventing `MerkleHashOrchard` from being used as a `HashMap` key. What motivated its original addition, and do we still depend on it downstream?

Reply:
Originally `incrementalmerkletree` was using hashmaps instead of btreemaps, but I changed that at Sean's request.

Reply:
Mmk. Then either document this removal in the changelog, or leave it in to allow other crate users who may continue to want to store `MerkleHashOrchard` in hashmaps.

Reply:
I think Sean's argument was that hashmaps generally shouldn't be used for cryptographic material, so I think removing it makes sense. This targets the non-consensus-changes branch, which doesn't have an up-to-date changelog, but I guess I can add it to the changelog here and then fix conflicts when that is merged down to `main`.

Reply:
@ebfull's rationale should be documented in the changelog removal entry